mos_tests/nfv/conftest.py

```diff
@@ -0,0 +1,120 @@
+                hp_computes.append(compute)
+    aggr = os_conn.nova.aggregates.create('hpgs-aggr', 'nova')
+    os_conn.nova.aggregates.set_metadata(aggr, {'hpgs': 'true'})
+    for host in hp_computes:
+        os_conn.nova.aggregates.add_host(aggr, host.data['fqdn'])
+    yield aggr
+    for host in hp_computes:
+        os_conn.nova.aggregates.remove_host(aggr, host.data['fqdn'])
+    os_conn.nova.aggregates.delete(aggr)
+
+
+@pytest.yield_fixture()
+def nfv_flavor(os_conn, clear_old, request):
+    flavors = getattr(
+        request, 'param', [[['m1.small.hpgs', 512, 1, 1],
+                           [{'hw:mem_page_size': 2048}, ]], ])
+    created_flavors = []
+
+    for flavor_params, flavor_keys in flavors:
+        flavor = os_conn.nova.flavors.create(*flavor_params)
+        for flavor_key in flavor_keys:
+            flavor.set_keys(flavor_key)
+        created_flavors.append(flavor)
+    yield created_flavors
+    for flavor in created_flavors:
+        os_conn.nova.flavors.delete(flavor.id)
+
+
+@pytest.yield_fixture(scope="class")
+def keypair(os_conn):
+    key = os_conn.create_key(key_name='nfv_key')
+    yield key.name
+    os_conn.delete_key(key_name=key.name)
+
+
+@pytest.yield_fixture(scope="class")
+def security_group(os_conn):
+    security_group = os_conn.create_sec_group_for_ssh()
+    yield security_group
+    os_conn.delete_security_groups()
+
+
+@pytest.yield_fixture(scope="class")
+def networks(os_conn):
+    router = os_conn.create_router(name="router01")['router']
+    ext_net = [x for x in os_conn.list_networks()['networks']
+               if x.get('router:external')][0]
+    os_conn.router_gateway_add(router_id=router['id'],
+                               network_id=ext_net['id'])
+    net01 = os_conn.add_net(router['id'])
+    net02 = os_conn.add_net(router['id'])
+    yield [net01, net02]
+
+    os_conn.delete_router(router['id'])
+    os_conn.delete_network(net01)
+    os_conn.delete_network(net02)
+
+
+@pytest.yield_fixture(scope="class")
+@pytest.yield_fixture
+def volume(os_conn):
+        image_id = [image.id for image in os_conn.nova.images.list()
+                    if image.name == 'TestVM'][0]
+        volume = common.create_volume(os_conn.cinder, image_id,
+                                      name='nfv_volume', type='volumes_lvm')
+        yield volume
+        volume.delete()
+
+
+@pytest.yield_fixture
+def clear_old(os_conn):
+    def clear(os_conn):
+        instance_list = os_conn.nova.servers.list()
+        if instance_list:
+            for instance in instance_list:
+                instance.delete()
+            common.wait(lambda: len(os_conn.nova.servers.list()) == 0,
+                        timeout_seconds=10 * 60)
+
+        image_list = os_conn.nova.images.list()
+        if image_list:
+            for image in image_list:
+                if image.name != 'TestVM':
+                    image.delete()
+            common.wait(lambda: len(os_conn.nova.images.list()) == 1,
```

Comment:
If ironic enabled, there is 4 images available by default.
Let's rewrite it like:
```python
@pytest.yield_fixture
def cleanup():
    instances = get_instances_list()
    images = get_images_list()
    yield
    delete_instances(get_instances_list() - instances)
    delete_images(get_images_list() - images)
```